src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -146,6 +151,13 @@ public override bool Execute()
                         UseShellExecute = false,
                         WorkingDirectory = Environment.CurrentDirectory,
                     };
+                    
+                    // Add environment variable to generate a restore binlog
+                    const string BinlogEnvironmentKey = "RESTORE_TASK_BINLOG_PARAMETERS";
```

Comment:
Looks good. How about adding unit test?

Reply:
https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/RestoreTaskExTests.cs or https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/MSBuildStaticGraphRestoreTests.cs are probably the right places to add a test.

Reply:
If you need set env var test sample then please check below. For setting env var value without affecting other unit test please:
check.https://github.com/NuGet/NuGet.Client/blob/73dac0ecc5172895c25d20b152e636504fb9751d/test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageArchiveReaderTests.cs#L1987-L2008